lib/p2p/Peer.ts

```diff
@@ -187,7 +189,7 @@ class Peer extends EventEmitter {
   /**
    * Close a peer by ensuring the socket is destroyed and terminating all timers.
    */
-  public close = (reason?: DisconnectingPacketBody): void => {
+  public close = (info?: DisconnectingPacketBody): void => {
```

Comment:
`info` is somewhat ambiguous here. Maybe instead of taking a `DisconnectingPacketBody` as a parameter, it can take `reason: DisconnectionReason` and `payload?: string`. I think that's a bit more clear and also more flexible in case we change what fields are used for the `DisconnectingPacketBody` type. If that makes sense to you then I think we'd want a similar method signature for `closePeer` below.